src/components/astro/AstroClassroomsTable.jsx

```diff
@@ -21,8 +21,15 @@ import {
   CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
 } from '../../ducks/classrooms';
 
-const AstroClassroomsTable = (props) => {
-  // TODO: Replace classroom.zooniverse_user_group_id with actual serialized user group id when available
+function calculateCompleteness(assignment, classroom) {
+  const classificationsTarget = +assignment.metadata.classifications_target;
+  const numberOfStudents = assignment.studentAssignments.length;
+  const numberOfCompletedClassifications = classroom.classificationsCount;
+  if (numberOfStudents === 0) return 0;
```

Comment:
Actually, all things considered, might it be a good idea in any case to change this to `if (numberOfStudents === 0 || classificationsTarget === 0) ...` to accommodate for those weird edge case Assignments? I'm not sure if the `assignment.classifications_target must be > 0` rule is enforced anywhere asides from the front end Assignment creation system.

(Or for the utterly paranoid, `if (!(numberOfStudents >= 1) || !(classificationsTarget >= 1) ...` )

Reply:
EDIT: I retract this comment, as this is just being too paranoid. If `classificationsTarget` is `0`, the worse that will most likely happen is that `Infinity` is displayed as a value instead of the whole app crashing.